### PR TITLE
Refactor/input helper

### PR DESCRIPTION
**Pull request by tomchentw**
## Depends on

~#24~

## Description

* Migrate blink timer for BBS char and cursor to toggle class on `document.body`
* Migrate inputHelper to bootstrap modal

## Screenshots

screen shot 2017-12-13 at 10 24 22 pm
screen shot 2017-12-13 at 10 24 16 pm
screen shot 2017-12-13 at 10 24 11 pm

### CODE REVIEW

**Comment by tomchentw** [comment]
Wow a simply `change base` would do the trick, awesome! @robertabcd ready for your review.

File: src/components/InputHelperModal.js [hunk 1]
```diff
@@ -0,0 +1,842 @@
...
+    };
+    const cached = {};
+    return {
+      onModalDialogMount: () => ref => {
+        if (ref) {
+          ref = ReactDOM.findDOMNode(ref);
+          ref.addEventListener("mousedown", onMouseDown);
+          ref.addEventListener("mousemove", onMouseMove);
+          ref.addEventListener("mouseup", onMouseUp);
+          cached.modalDialog = ref;
```
Comment (robertabcd) [inline src/components/InputHelperModal.js]: Can this be deleted? Same as below.

**Review by robertabcd — COMMENTED** [review]
I didn't look closely, nor am I familiar with this part of code. I might over look things here.

File: src/components/InputHelperModal.js (hunk 1 above)
Comment (tomchentw) [reply src/components/InputHelperModal.js]: Yes, this is redundent

Comment (tomchentw) [reply src/components/InputHelperModal.js]: Fixed now.

**Comment by tomchentw** [comment]
Fixed the code review comment, thanks!

Would someone else (maybe the author _iamchucky_?) be able to help review this part of code?

**Comment by tomchentw** [comment]
Is there anything I can do to help get this reviewed? Basically, what I did is:

* Play around current input helper module, try its behavior
* Create the same UI with React, in the same time removing the UI generation part of code from the input helper module
* Then, the rest piece of code in the input helper module would be event listeners. Just hook them up to react & the existing system.

I have two following PRs that refactors out the modals to React, and a working branch to reactor the context menu as a whole to React in pending. I believe that should extract the major case of the application logic (`pttchrome` module) to React, and helps the further refactoring for:

* Move current UI creation/event listeners to the top-level React component
* Then, the rest piece of code would be only telnet/buf, which is the most complicated part

**Comment by robertabcd** [comment]
Sorry for the misunderstanding. I was going to merge after my comment get addressed. I didn't mean I need other reviewers for review.

**Comment by hdd60311** [comment]
回報一個bug就是在firefox上開啟輸入小幫手後點右上角x紐無法關閉，chrome上沒這問題

**Comment by tomchentw** [comment]
@hdd60311 謝謝，我來看一下

---

Fixed by #28

**Pull request merged**